app/gui/src/dashboard/pages/dashboard/Dashboard.tsx

```diff
@@ -308,3 +316,130 @@ function DashboardInner(props: DashboardProps) {
+                continue
+              }
+              if (directoriesInCategory.includes(currentParentId)) {
+                continue
+              }
+              const id = localBackendModule.newDirectoryId(currentParentPath)
+              // This is SAFE as the value has been `structuredClone`d above.
+              unsafeMutable(directoriesInCategory).push(id)
+            }
+          }
+        }
+      }
+    }
+
+    const cloudProjects = launchedProjects.filter(
+      (project) => project.type === backendModule.BackendType.remote,
+    )
+    const promises = cloudProjects.map((project) =>
+      queryClient.ensureQueryData(
+        listDirectoryQueryOptions({
+          backend: remoteBackend,
+          parentId: project.parentId,
+          category,
+        }),
+      ),
+    )
+    const projectsSiblings = await Promise.allSettled(promises)
+    const projects = projectsSiblings.flatMap((directoryResult, i) => {
+      const projectInfo = cloudProjects[i]
+      const project =
+        projectInfo && directoryResult.status === 'fulfilled' ?
+          directoryResult.value
+            .filter(backendModule.assetIsProject)
+            .find((asset) => asset.id === projectInfo.id)
+        : null
+      return project ? [project] : []
+    })
+    for (const project of projects) {
+      const parents = project.parentsPath.split('/').filter(backendModule.isDirectoryId)
+      const rootDirectoryId = parents[0]
+      const baseVirtualPath = (() => {
+        const userGroupName = userGroups?.find(
+          (userGroup) => userGroupIdToDirectoryId(userGroup.id) === rootDirectoryId,
+        )?.groupName
+        if (userGroupName != null) {
+          return `enso://Teams/${userGroupName}`
+        }
+        const userName = users?.find(
+          (otherUser) => userIdToDirectoryId(otherUser.userId) === rootDirectoryId,
+        )?.name
+        if (userName != null) {
+          return `enso://Users/${userGroupName}`
+        }
+      })()
+      const virtualPath = backendModule.Path(`${baseVirtualPath}/${project.virtualParentsPath}`)
+      invariant(
+        baseVirtualPath != null,
+        'The root directory must be either a user directory or a team directory.',
+      )
+      for (const [categoryRootPath, directoriesInCategoryRaw] of unsafeEntries(
+        expandedDirectories,
+      )) {
+        const directoriesInCategory = unsafeMutable(directoriesInCategoryRaw)
+        if (!backendModule.isDescendantPath(virtualPath, categoryRootPath)) {
+          continue
+        }
+        if ((rootDirectoryId && userGroupDirectoryIds.has(rootDirectoryId)) ?? false) {
+          directoriesInCategory.push(TEAMS_DIRECTORY_ID)
+        } else {
+          directoriesInCategory.push(USERS_DIRECTORY_ID)
+        }
+        for (const parent of parents) {
+          directoriesInCategory.push(parent)
+        }
+      }
+    }
+
+    setExpandedDirectories(expandedDirectories)
+  })
+
+  React.useEffect(() => {
```

Comment:
Why we use `useEffect` here, instead of setting default values in the store?

Reply:
as mentioned on discord a couple days ago: we can do that but this code depends on a couple of other stores (see the start of the function)

i feel like putting this code in the store will make the store too tightly coupled to the other stores, but if you think that's fine then i'm happy to make the change

Reply:
We can pass the stores as props and keep them decoupled

Reply:
Your code couples store but in unobvious way

Reply:
> setting default values in the store

small problem: fetching parent directories for cloud projects is async

Reply:
I'm about splitting, not fetching. assume the data is present

Reply:
> small problem: fetching parent directories for cloud projects is async

We have suspense for that, haven't we?